astropy/io/ascii/tests/test_qdp.py

```diff
@@ -0,0 +1,135 @@
+import numpy as np
+import pytest
+from astropy.io.ascii import core
+from astropy.io.ascii.qdp import read_table_qdp, write_table_qdp
+from astropy.table import Table, Column, MaskedColumn
+from astropy.utils.exceptions import AstropyUserWarning
+
+
+def test_get_tables_from_qdp_file(tmpdir):
+    example_qdp = """
+    ! Swift/XRT hardness ratio of trigger: XXXX, name: BUBU X-2
+    ! Columns are as labelled
+    READ TERR 1
+    READ SERR 2
+    ! WT -- hard data
+    !MJD            Err (pos)       Err(neg)        Rate            Error
+    53000.123456 2.37847222222222e-05    -2.37847222222222e-05   -0.212439       0.212439
+    55045.099887 1.14467592592593e-05    -1.14467592592593e-05   0.000000        0.000000
+    NO NO NO NO NO
+    ! WT -- soft data
+    !MJD            Err (pos)       Err(neg)        Rate            Error
+    53000.123456 2.37847222222222e-05    -2.37847222222222e-05   0.726155        0.583890
+    55045.099887 1.14467592592593e-05    -1.14467592592593e-05   2.410935        1.393592
+    NO NO NO NO NO
+    ! WT -- hardness ratio
+    !MJD            Err (pos)       Err(neg)        Rate            Error
+    53000.123456 2.37847222222222e-05    -2.37847222222222e-05   -0.292553       -0.374935
+    55045.099887 1.14467592592593e-05    -1.14467592592593e-05   0.000000        -nan
+    """
+
+    path = str(tmpdir.join('test.qdp'))
+
+    with open(path, "w") as fp:
+        print(example_qdp, file=fp)
+
+    table0 = read_table_qdp(fp.name, names=["MJD", "Rate"], table_id=0)
+    assert table0.meta["initial_comments"].lstrip().startswith("Swift")
+    assert table0.meta["comments"].lstrip().startswith("WT -- hard data")
```

Comment:
Comments should be a list instead of string. See https://docs.astropy.org/en/stable/io/ascii/read.html#comments-and-metadata.

Reply:
Done